From 2d017b5bf6f4c42126ddc1b2f60a5a397643dcd5 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Tue, 22 Mar 2022 13:30:04 +0200
Subject: [PATCH] [SW-0] gaudi3: rom: fix status update

The logic for SW status update for bringup is hardcoded to
SW_STATUS_0 register. As in gaudi3 the number of status flags
have increased, the image bringup statuses cross 32. Due to
this we need to update the logic to access SW_STATUS_1 registers
and so on according to the status bit number.

Change-Id: Ic1dde5a230a734f6f3034f56892ccbf6b557a211
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 93c752f..44e8192 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -58,11 +58,7 @@
 void arcboot_rom_status_update_bringup(enum arcboot_rom_progress progress,
 					      enum arcboot_rom_bringup_proggress position)
 {
-	uint32_t reg;
-
-	reg = regmap_readl(mmSW_STATUS_0);
-	reg |= BIT(progress + position);
-	regmap_writel(reg, mmSW_STATUS_0);
+	arcboot_rom_status_update(progress + position);
 }
 
 void arcboot_rom_status_update_location(uint8_t progress,
